newrelic-weaver/src/main/java/com/newrelic/weave/weavepackage/WeavePackage.java

```diff
@@ -142,8 +142,25 @@ public WeavePackage(WeavePackageConfig config, List<byte[]> weavePackageBytes) {
                     requiredClassAnnotationsLookup.put(weaveNode.name, weave.getRequiredClassAnnotations());
                 }
                 if (isMethodAnnotationMatch) {
```

Comment:
Not sure if it is worth adding similar logic to the above condition when `isClassAnnotationMatch` is `true`? I was trying to be minimally invasive and currently there don't appear to be any instrumentation modules that have a similar situation with class annotations but technically the same issue could happen there.